### DISCUSSION TITLE
Clean smart cards not detected by OpenSC

### DISCUSSION THREAD

**Discussion by paulober** [OpenSC/OpenSC#2994] [Q&A] [answered]
I have two readers one expensive German authorities certified one an a cheap one. I also got two types of smart cards SLE4428 and SLE4442. If I plug them into the reader(s), macOS shows "Smart Card detected" (generic smart card reader interface).
But neither `sudo security list-smartcards` nor `sudo opensc-tool -n 0`/`opensc-explorer` work.
The output of `opensc-tool --list-readers ` is:
```
# Detected readers (pcsc)
Nr.  Card  Features  Name
0    Yes             Generic Smart Card Reader Interface
```

Can anyone help me how I can write to this cards?

**Comment by popovec** [comment, marked as answer]
https://github.com/OpenSC/OpenSC/issues/237

SLE 4418 / SLE 4428 - Intelligent 8-Kbit EEPROM 

This is out of OpenSC scope .. this card does not have RSA or EC cryptographic operations implemented - it cannot be used for digital signature/encryption.

**Reply by paulober** [reply]
Thanks for the quick reply.
I can't find any software which I can use to write to this cards. Some explanations say I have to send `APDU` commands to the card. So I thought opensc-tool -s could be used for that, but it did not work.

**Reply by popovec** [reply]
According to the documentation, it is a synchronous card and has no support for APDU. It's just a serial EEPROM.

**Comment by frankmorgner** [comment]
please note that opensc-notify sends "Smart Card Detected" purely based on the PC/SC event, which - as stated above - also applies to cards, that are not supported by OpenSC.
